test/cluster/test_multidc.py

```diff
@@ -464,3 +465,127 @@ async def test_restart_with_prefer_local(request: pytest.FixtureRequest, manager
+
+    # Setup: DC1: 3 racks, DC2: 2 racks.
+    start_node_tasks = [
+        manager.server_add(config=cfg, property_file={"dc": "dc1", "rack": "r1"}),
+        manager.server_add(config=cfg, property_file={"dc": "dc1", "rack": "r2"}),
+        manager.server_add(config=cfg, property_file={"dc": "dc1", "rack": "r3"}),
+        manager.server_add(config=cfg, property_file={"dc": "dc2", "rack": "r4"}),
+        manager.server_add(config=cfg, property_file={"dc": "dc2", "rack": "r5"}),
+    ]
+    await asyncio.gather(*start_node_tasks)
+
+    cql = manager.get_cql()
+
+    # Scenario 1. Creating an RF-rack-invalid keyspace.
+    ###################################################
+
+    async def do_create_test(rf1: int, rf2: int, tablets: str, ok: bool):
+        ks = unique_name()
+        warning = f"The keyspace '{ks}' you're trying to create doesn't satisfy the requirements " \
+                   "to be RF-rack-valid, i.e. it uses tablets, but the replication factor in at least " \
+                   "one of the data centers does not match the number of racks in that data center. " \
+                   "That may result in worse availability. Consider updating the replication factors " \
+                   "to satisfy that."
+
+        stmt = f"CREATE KEYSPACE {ks} WITH replication = " \
+               f"{{'class': 'NetworkTopologyStrategy', 'dc1': {rf1}, 'dc2': {rf2}}} AND " \
+               f"tablets = {{'enabled': {tablets}}}"
+
+        # We have to use `Session::execute_async` here to be able to obtain `warnings`.
+        # It's pretty convoluted, but we have to live with it...
+        result = cql.execute_async(stmt)
+        await _wrap_future(result)
+
+        if ok:
+            assert not hasattr(result, "warnings") or result.warnings is None or warning not in result.warnings
+        else:
+            assert hasattr(result, "warnings")
+            assert warning in result.warnings
+
+    tasks = [
+        # All of the statements below are OK: they don't use tablets.
+        do_create_test(2, 2, "false", True),
+        do_create_test(3, 3, "false", True),
+        do_create_test(3, 2, "false", True),
+        do_create_test(3, 1, "false", True),
+        do_create_test(1, 2, "false", True),
+        do_create_test(1, 1, "false", True),
+
+        # BAD: the RF doesn't match the number of racks in DC1.
+        do_create_test(2, 2, "true", False),
+        # BAD: the RF doesn't match the number of racks in DC2.
+        do_create_test(3, 3, "true", False),
+        # OK: the RFs match the number of racks in the DCs.
+        do_create_test(3, 2, "true", True),
+        # OK: RF=#racks for DC1, RF=1 is always accepted.
+        do_create_test(3, 1, "true", True),
+        # OK: RF=#racks for DC2, RF=1 is always accepted.
+        do_create_test(1, 2, "true", True),
+        # OK: RF=1 is always accepted.
+        do_create_test(1, 1, "true", True),
+    ]
+    await asyncio.gather(*tasks)
```

Comment:
I'm not sure why we have to go to such lengths to run those checks in parallel. After all, they only try to create a keyspace and check that they got an error - it should be rather fast, no?